### ISSUE TITLE
C++ visualizer problem with C-String literals

### ISSUE THREAD

**Issue by stephendgilbert** [pgbovine/opt-cpp-backend#64]
Hello,
I teach second semester C++.  I've had some (good) students getting one of the quiz questions wrong. I just discovered it was because they were using the visualizer and it was giving them the wrong information. The words "Goodbye" are not on the heap in this problem, but in the read-only portion of the static storage area. 
https://goo.gl/vn1b1U

**Comment by pgbovine** [comment]
Good point; unfortunately I don't have the resources right now to fix this (since the underlying implementation doesn't easily expose those details). I'll leave this issue open so that people can see it.

**Comment by DaveParillo** [comment]
Perhaps related: a simple example using std::string, seems to lose track of the pointers.

https://goo.gl/9F8NTo

The execution shows the pointers have values, but the visualization shows poop.
